crates/cubecl-std/src/quant/view.rs

```diff
@@ -218,6 +218,55 @@ impl<Q: CubePrimitive, S: CubePrimitive, F: Float, C: Coordinates + 'static>
     }
 }
 
+struct ExpandDynamic<'a, E: Numeric, C: Coordinates + 'static> {
+    values: &'a ViewCompilationArg<C>,
+    scales: &'a ViewCompilationArg<C>,
+    scheme: QuantScheme,
+    builder: &'a mut KernelBuilder,
+    _ty: PhantomData<E>,
+}
+
+impl<'a, E: Numeric, C: Coordinates + 'static> RunWithQuantType for ExpandDynamic<'a, E, C> {
+    type Output = ViewExpand<Line<E>, C>;
+
+    fn execute<Q: CubePrimitive, S: CubePrimitive>(self) -> Self::Output {
+        let values = View::<Line<Q>, C>::expand(self.values, self.builder);
+        let scales = View::<S, C>::expand(self.scales, self.builder);
+        let view = QuantizedViewExpand::new(values, scales, self.scheme);
+        ViewExpand::new(view)
+    }
+}
+
+pub fn run_with_quant_type<F: RunWithQuantType>(func: F, scheme: QuantScheme) -> F::Output {
```

Comment:
This would benefit from more docs.

Reply:
Done. It's mainly for fusion since that also needs to dynamically instantiate the views with the right types. But could be useful in the future.